java-frontend/src/main/java/org/sonar/java/model/expression/InstanceOfTreeImpl.java

```diff
@@ -42,21 +43,28 @@ public class InstanceOfTreeImpl extends AssessableExpressionTree implements Inst
   private final TypeTree type;
   @Nullable
   private final VariableTree variable;
+  @Nullable
+  private final PatternTree pattern;
```

Comment:
I'm not sure we need to keep the `VariableTree` here. What about keeping only the pattern?

For me, the case `variable != null`  corresponds to the situation where the use is a  `TypePattern`.

Dropping the variable would prevent the introduction of yet another constructor.